pkg/apis/core/v1/storagecluster.go

```diff
@@ -531,9 +531,9 @@ type DataProviderSpec struct {
 
 // GitOpsSpec contains the details for GitOps provider like github or bitbucket
 type GitOpsSpec struct {
-	Name   string                 `json:"name,omitempty"`
-	Type   string                 `json:"type"`
-	Params map[string]interface{} `json:"params"`
+	Name   string            `json:"name,omitempty"`
+	Type   string            `json:"type"`
+	Params map[string]string `json:"params"`
```

Comment:
add omitempty as discussed? do we need omitempty for type as well? cc: @piyush-nimbalkar

Reply:
Oh yes. Let's add omitempty for both `Type` and `Params`.
Without it the StorageCluster spec looks unnecessarily long.

Reply:
we don't want the omitemty as this config is not usable w/o params. The whole point of the `omitempty` as I understand is allowing to proceed w/o this field. We can't let user create gitops type and omit params

Reply:
That's not the purpose of `omitempty`. It only tells the yaml parser to omit the field if it is set to a default value based on the type of the field. For instance, it will not show the fields in the yaml for an empty map, empty string, false bool, 0 int, etc
It is not same as 'required', where it will stop the user from proceeding w/o this field.